pkg/local_object_storage/engine/evacuate.go

```diff
@@ -138,13 +146,17 @@ func (e *StorageEngine) Evacuate(prm EvacuateShardPrm) (EvacuateShardRes, error)
 				}
 			}
 
-			// TODO (@fyrchik): #1731 try replicating to another node.
-			//  The suggestion is to have prm.handler which is called
-			//  if a Put has failed.
+			if prm.handler == nil {
+				// Do not check ignoreErrors flag here because
+				// ignoring errors on put make this command kinda useless.
+				return res, fmt.Errorf("%w: %s", errPutShard, lst[i])
+			}
 
-			// Do not check ignoreErrors flag here because
-			// ignoring errors on put make this command kinda useless.
-			return res, fmt.Errorf("%w: %s", errPutShard, lst[i])
+			err = prm.handler(lst[i], getRes.Object())
+			if err != nil {
+				return res, err
+			}
+			res.count++
```

Comment:
so technically that that counter is not about the object being evacuated successfully now OR the handler MUST evacuate the object somewhere?

Reply:
Fixed a comment for `Count`